### PR TITLE
CAS-698: Fix Voting Success Modal appearing too early

### PR DESCRIPTION
**Pull request by jbluks**

### CODE REVIEW

File: frontend/packages/client/src/pages/Proposal.js [hunk 1]
```diff
@@ -298,6 +299,14 @@ export default function ProposalPage() {
...
   }
 
+  console.log(
+    confirmingVote,
+    castingVote,
+    castVote,
+    String(castVote),
+    confirmingVote && castingVote && String(castVote)
+  );
+
```
Comment (germanurrus) [inline frontend/packages/client/src/pages/Proposal.js:309]: remove output

File: frontend/packages/client/src/pages/Proposal.js [hunk 2]
```diff
@@ -162,6 +162,7 @@ export default function ProposalPage() {
   };
 
   const onConfirmVote = (options) => {
+    console.log('onConfirmVote', options);
```
Comment (germanurrus) [inline frontend/packages/client/src/pages/Proposal.js:165]: remove output

File: frontend/packages/client/src/pages/Proposal.js [hunk 3]
```diff
@@ -325,7 +334,7 @@ export default function ProposalPage() {
           </div>
         </div>
       )}
-      {confirmingVote && castingVote && String(castVote) && (
+      {confirmingVote && castingVote && castVote && (
```
Comment (germanurrus) [inline frontend/packages/client/src/pages/Proposal.js:337]: maybe doing `castVote !== null` since it's initialised with null ? an option could be a 0

**Review by germanurrus — APPROVED** [review]

File: frontend/packages/client/src/pages/Proposal.js (hunk 3 above)
Comment (jbluks) [reply frontend/packages/client/src/pages/Proposal.js]: Yes - that is the correct fix here.

**Pull request merged**